tenants-certified-attributes-updater/src/main/scala/it/pagopa/interop/tenantscertifiedattributesupdater/Dependencies.scala

```diff
@@ -0,0 +1,163 @@
+import it.pagopa.interop.commons.signer.service.SignerService
+import it.pagopa.interop.commons.signer.service.impl.KMSSignerService
+import it.pagopa.interop.commons.utils.TypeConversions.TryOps
+import it.pagopa.interop.partyregistryproxy.client.api.InstitutionApi
+import it.pagopa.interop.partyregistryproxy.client.model.{Institution, Institutions}
+import it.pagopa.interop.tenantmanagement.model.tenant.{PersistentExternalId, PersistentTenant}
+import it.pagopa.interop.tenantprocess.client.api.TenantApi
+import it.pagopa.interop.tenantprocess.client.model.{ExternalId, InternalAttributeSeed, InternalTenantSeed}
+import it.pagopa.interop.tenantscertifiedattributesupdater.service.impl.{
+  PartyRegistryProxyServiceImpl,
+  TenantProcessServiceImpl
+}
+import it.pagopa.interop.tenantscertifiedattributesupdater.service.{PartyRegistryProxyInvoker, TenantProcessInvoker}
+import it.pagopa.interop.tenantscertifiedattributesupdater.system.ApplicationConfiguration
+import it.pagopa.interop.tenantscertifiedattributesupdater.util.{AttributeInfo, TenantActions}
+import org.mongodb.scala.MongoClient
+
+import java.util.UUID
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutor, Future}
+import scala.util.Try
+
+trait Dependencies {
+
+  implicit val contexts: Seq[(String, String)] = Seq.empty
+
+  final val initPage: Int       = 1
+  final val maxLimit: Int       = 100
+  final val groupDimension: Int = 1000
+
+  val jwtConfig: JWTInternalTokenConfig = JWTConfiguration.jwtInternalTokenConfig
+
+  def partyRegistryProxyService(
+    blockingEc: ExecutionContextExecutor
+  )(implicit actorSystem: ActorSystem[_]): PartyRegistryProxyServiceImpl =
+    PartyRegistryProxyServiceImpl(
+      PartyRegistryProxyInvoker(blockingEc)(actorSystem.classicSystem),
+      InstitutionApi(ApplicationConfiguration.partyRegistryProxyURL)
+    )(blockingEc)
+
+  def tenantProcessService(
+    blockingEc: ExecutionContextExecutor
+  )(implicit actorSystem: ActorSystem[_]): TenantProcessServiceImpl =
+    TenantProcessServiceImpl(
+      TenantProcessInvoker(blockingEc)(actorSystem.classicSystem),
+      TenantApi(ApplicationConfiguration.tenantProcessURL)
+    )(blockingEc)
+
+  def signerService(blockingEc: ExecutionContextExecutor): SignerService = new KMSSignerService(blockingEc)
+
+  def generateBearer(
+    blockingEc: ExecutionContextExecutor
+  )(implicit actorSystem: ActorSystem[_], ec: ExecutionContext): Future[String] = for {
+    tokenGenerator <- interopTokenGenerator(blockingEc)
+    m2mToken       <- tokenGenerator
+      .generateInternalToken(
+        subject = jwtConfig.subject,
+        audience = jwtConfig.audience.toList,
+        tokenIssuer = jwtConfig.issuer,
+        secondsDuration = jwtConfig.durationInSeconds
+      )
+  } yield m2mToken.serialized
+
+  def interopTokenGenerator(
+    blockingEc: ExecutionContextExecutor
+  )(implicit actorSystem: ActorSystem[_], ec: ExecutionContext): Future[InteropTokenGenerator] =
+    Try(
```

Comment:
The constructor of Future already traps exceptions. There's no need to resort using a Try.